### PR TITLE
Add optional go compatibility unit and integration presubmits

### PR DESCRIPTION
**Pull request by liggitt**
Related to https://github.com/kubernetes/test-infra/issues/28310 and https://github.com/kubernetes/enhancements/issues/3744

* Clones the unit/integration presubmits for release branches (first commit)
* Makes them optional and non-blocking, and records the GO_VERSION that release branch originally shipped with (second commit)

Eventually intended to be used in combination with something like https://github.com/kubernetes/kubernetes/pull/115377

Once this is proven out, we can figure out how to keep these created / updated as part of the fork-config script

cc @dims @BenTheElder

### CODE REVIEW

**Comment by dims** [comment]
needs a couple of annotations @liggitt:
https://cs.k8s.io/?q=testgrid-dashboards%7Ctestgrid-tab-name&i=nope&files=&excludeFiles=&repos=kubernetes/test-infra

**Comment by liggitt** [comment]
> needs a couple of annotations @liggitt:

ack, put in the sig-testing-canaries dashboard alongside the go-canary jobs for now

File: config/jobs/kubernetes/sig-release/release-branch-jobs/1.24.yaml [hunk 1]
```diff
@@ -1420,6 +1420,38 @@ presubmits:
...
+    path_alias: k8s.io/kubernetes
+    spec:
+      containers:
+      - args:
+        - ./hack/jenkins/test-dockerized.sh
+        command:
+        - runner.sh
+        env:
+        - name: GO_VERSION
+          value: "1.18.1"
```
Comment (cpanato) [inline config/jobs/kubernetes/sig-release/release-branch-jobs/1.24.yaml:1443]: any specific reason to use an older version?

Comment (cpanato) [reply config/jobs/kubernetes/sig-release/release-branch-jobs/1.24.yaml:1443]: ahh i see now, but did not get it why.

**Comment by liggitt** [comment]
triggered green runs of these on release-branch PRs with backport of the `gimme` go change that made GO_VERSION effective:
* 1.26: https://github.com/kubernetes/kubernetes/pull/115496
* 1.25: https://github.com/kubernetes/kubernetes/pull/115497
* 1.24: https://github.com/kubernetes/kubernetes/pull/115498
* 1.23: https://github.com/kubernetes/kubernetes/pull/115499

**Pull request merged**